pkg/apis/experimental/types.go

```diff
@@ -639,3 +639,36 @@ type ClusterAutoscalerList struct {
 
 	Items []ClusterAutoscaler `json:"items"`
 }
+
+// Selector is a label query over a set of API objects. MatchLabels and
+// MatchExpressions are ANDed. An empty Selector matches all objects. A
+// null Selector matches no objects.
+type Selector struct {
+	// MatchLabels is a list of labels.
+	MatchLabels map[string]string `json:"matchLabels,omitempty"`
+	// MatchExpressions is a list of requirement selectors.
```

Comment:
The comment is ambiguous: it says "MatchLabels and MatchExpressions are ANDed" but it doesn't describe how the elements within a MatchExpression are combined. I assumed AND, but the comment should be clarified.

Unlike numerical comparisons and "not in" (comments below), I don't feel very strongly about allowing OR here at the moment, but would like to think a little more to see if I can think of realistic use cases. I assume the objection is the "hard to detect overlap" issue you mentioned later?